src/main/java/org/mskcc/cmo/messaging/impl/JSGatewayImpl.java

```diff
@@ -297,4 +298,48 @@ public PublishingQueueTask(String msgId, String subject, Object payload) {
             this.payload = payload;
         }
     }
+
+    @Override
+    public Message request(String subject, Object message) {
+        if (!isConnected()) {
+            throw new IllegalStateException("Gateway connection has not been established.");
+        }
+        try {
+            String msg = mapper.writeValueAsString(message);
+            Message reply = natsConnection.request(subject, msg.getBytes(),
+                    Duration.ofSeconds(requestWaitTime));
+            if (reply == null) {
+                LOG.error("No reply received for a request using NATS connection");
+            } else {
+                return reply;
+            }
+        } catch (Exception ex) {
+            LOG.error("Error during attempt to send a request using NATS connection", ex);
+        }
+        return null;
+    }
+
+    @Override
+    public void reply(String subject, Object message, MessageConsumer messageConsumer) {
```

Comment:
after tinkering around with this code a bit I'm not sure we need to pass a message consumer to this method since it's not going to be used here because a new one is created when setting up the dispatcher. 

I'll follow up later after I've been able to set up a java implementation for request-reply in the metadb service